src/pluggy/_callers.py

```diff
@@ -15,46 +15,47 @@ def _multicall(hook_name, hook_impls, caller_kwargs, firstresult):
     __tracebackhide__ = True
     results = []
     excinfo = None
-    try:  # run impl and wrapper setup functions in a loop
-        teardowns = []
-        try:
-            for hook_impl in reversed(hook_impls):
-                try:
-                    args = [caller_kwargs[argname] for argname in hook_impl.argnames]
-                except KeyError:
-                    for argname in hook_impl.argnames:
-                        if argname not in caller_kwargs:
-                            raise HookCallError(
-                                f"hook call must provide argument {argname!r}"
-                            )
-
-                if hook_impl.hookwrapper:
-                    try:
-                        gen = hook_impl.function(*args)
-                        next(gen)  # first yield
-                        teardowns.append(gen)
-                    except StopIteration:
-                        _raise_wrapfail(gen, "did not yield")
-                else:
-                    res = hook_impl.function(*args)
-                    if res is not None:
-                        results.append(res)
-                        if firstresult:  # halt further impl calls
-                            break
-        except BaseException:
```

Comment:
I think I've already stated this, but just to re-iterate: IMO instead this PR should change this to `Exception` and we shouldn't be handling `BaseException` at all (unless we need it for `GeneratorExit`).